src/utils/deviceInfo/deviceInfo.ts

```diff
@@ -0,0 +1,170 @@
+        this.orientationIsLandscapeQuery.addEventListener('change', () => (this._orientation = this.getOrientation()));
+    }
+
+    private getUpdatedScreenDimensions(): ScreenDimensions {
+        const updatedDimensions = {
+            width: window.innerWidth,
+            height: window.innerHeight,
+            visual: window.visualViewport
+                ? {
+                      width: window.visualViewport.width,
+                      height: window.visualViewport?.height
+                  }
+                : undefined,
+            fullscreen: window.screen
+                ? {
+                      available: {
+                          width: window.screen.availWidth,
+                          height: window.screen.availHeight
+                      },
+                      total: {
+                          width: window.screen.width,
+                          height: window.screen.height
+                      }
+                  }
+                : undefined
+        } as ScreenDimensions;
+        this._dimensions = updatedDimensions;
+        return updatedDimensions;
+    }
+
+    private getBrowser(): Browser | undefined {
+        const browser = this._detailedDeviceInfo.deviceInformation.webBrowser.toLowerCase();
+
+        if (browser.includes('chrome') || browser.includes('chromium')) return 'Chrome';
+        if (browser.includes('firefox')) return 'Firefox';
+        if (browser.includes('opera')) return 'Opera';
+        if (browser.includes('safari')) return 'Safari';
+        if (browser.includes('edge')) return 'Edge';
+
+        return undefined;
+    }
+
+    private getUserInput() {
+        const usingTouchQuery = window.matchMedia('(pointer: coarse)');
+        return usingTouchQuery.matches ? 'touch' : 'mouse';
+    }
+
+    /** Returns the current viewing medium.*/
+    private getDeviceType(): DeviceType {
+        const deviceInfo = this._detailedDeviceInfo.deviceInformation;
+
+        // Find iDevices
+        if (deviceInfo.deviceModel === 'iPad Apple') return 'tablet';
+        if (deviceInfo.deviceModel === 'iPhone Apple') return 'mobile';
+        if (deviceInfo.platform === 'MacOs') return 'desktop';
+
+        // Find Android phones.
+        if (deviceInfo.platform === 'Android') return 'mobile';
+
+        /* Find Windows desktops.
+         * Implementers should note users will have touch screens on Windows desktop devices. */
+        if (deviceInfo.platform === 'Windows') return 'desktop';
+
+        /* Find Linux desktops with mouse inputs.
+         * A clash with Android mobile or tablet is a possibility. */
+        if (deviceInfo.platform === 'Linux' && this._userInput === 'mouse') return 'desktop';
+
+        return 'desktop';
+    }
+
+    private getOrientation(): ScreenOrientation {
+        if (this.orientationIsLandscapeQuery.matches) return 'landscape';
+        else return 'portrait';
+    }
+
+    public getDeviceDetails(): DetailedDeviceInformation {
```

Comment:
Methods should be `public` by default, so no need to add `public` keyword here, and the other places.